src/components/data_display/routes.ts

```diff
@@ -152,24 +156,80 @@ const routeDataViewFilterList = async (request: any, reply: any) => {
 
   const dataView = dataViews[0]
 
-  // Check column exists on highest priority dataView
-  if (!dataView.tableViewIncludeColumns?.includes(columnName))
-    throw new Error(`Column "${dataViewCode}" not available on data view: "${dataView.title}"`)
+  // TO-DO: Create search filters for types other than string (number, bool, array)
+  const searchFilter =
+    searchText === ''
+      ? {}
+      : {
+          or: searchFields.map((col: string) => ({
+            [col]: {
+              includesInsensitive: searchText,
+            },
+          })),
+        }
+
+  const gqlFilters = { ...getFilters(dataView, userId, orgId), ...searchFilter }
+
+  const filterList = new Set()
 
-  // GraphQL query column including row_restrictions
-  const gqlFilters = getFilters(dataView, userId, orgId)
+  let fetchedCount = 0
+  let offset = 0
+  let moreResultsAvailable = true
 
-  const items = (
-    await queryFilterList(
+  while (filterList.size < config.filterListMaxLength) {
+    const { fetchedRecords, totalCount, error } = await queryFilterList(
       camelCase(getValidTableName(dataView.tableName)),
-      columnName,
+      searchFields,
       gqlFilters,
+      config.filterListMaxLength,
+      offset,
       authHeaders
     )
-  ).map((item: { [key: string]: unknown }) => item[columnName])
 
-  // TO-DO: How to handle very large lists?
-  return reply.send([...new Set(items)])
+    if (error) return reply.send(error)
+
+    fetchedCount += fetchedRecords.length
+
+    fetchedRecords.forEach((record: { [key: string]: unknown }) => {
+      // For some reason, if a single field value is `null`, Postgraphile
+      // returns null instead of an object with a null value on one field, like
+      // the rest of the results
+      if (record === null) {
+        filterList.add(null)
+        return
+      }
+      const values = Object.values(record)
+      values.forEach((value) => {
+        if (delimiter && typeof value === 'string') {
+          const splitString = value.split(delimiter).map((e) => e.trim())
+          splitString.forEach((string) => filterList.add(string))
+        } else filterList.add(value)
+      })
+    })
+
+    if (fetchedCount >= totalCount) {
+      moreResultsAvailable = false
+      break
+    }
+
+    offset += fetchedRecords.length
+  }
+
+  let results = [...filterList].sort()
+
+  // If searching multiple fields, then we'll also get back the *other* values
+  // in the record matching the search text. We need to filter those out:
+  if (searchFields.length > 1)
+    results = results.filter(
+      (value) =>
+        (typeof value === 'string' &&
+          new RegExp(searchText.toLowerCase()).test(value.toLowerCase())) ||
+        value === null
+    )
+
+  if (results.length > config.filterListMaxLength) moreResultsAvailable = true
+
+  return reply.send({ list: results.slice(0, config.filterListMaxLength), moreResultsAvailable })
```

Comment:
I know that most cases `filterListMaxLength` is defined, but it would be good for safe-guard to have a local const that gets this or something else defined in here.

Reply:
It's hard-coded in config.ts, so shouldn't be missing. But yeah, due to the typing of the config object, it doesn't actually throw an error when a property is missing. We should really rectify that at some point, but for now I've just added a fallback as you suggested.

c83a5827